src/Sarif/EnumeratedArtifact.cs

```diff
@@ -64,6 +66,39 @@ public byte[] Bytes
             set => this.bytes = value;
         }
 
+        public string CleanPath
+        {
+            get => GetCleanFilePath(Uri.OriginalString);
+            set => this.cleanPath = value;
+        }
+
+        private readonly static Regex AbsoluteFilePathWithFileSchemeRegex = new Regex(@"file:(?:\/{3}|\/{2}\w+\/|\/{1})?(.+)", RegexOptions.Compiled | RegexOptions.CultureInvariant);
+
+        private string GetCleanFilePath(string originalString)
+        {
+            if (this.cleanPath != string.Empty)
+            {
+                return this.cleanPath;
+            }
+
+            if (originalString.StartsWith("file:"))
+            {
+                Match match = AbsoluteFilePathWithFileSchemeRegex.Match(originalString);
+                if (match.Success)
+                {
+                    string separator = Path.DirectorySeparatorChar.ToString();
+                    this.cleanPath = match.Groups[1].Value.Replace("/", separator);
+                    return this.cleanPath;
```

Comment:
An example is as follows:
If the originalString is **file:///C:/Users/v-zhouling/AppData/Local/Temp/New%2DYapeAzureCertificateForWinRMOverHttpsInKeyVault.md**,
the transferred string (in Windows) will be: **C:\Users\v-zhouling\AppData\Local\Temp\New%2DYapeAzureCertificateForWinRMOverHttpsInKeyVault.md**

Reply:
Isn't that `%2D` a url-encoded `-` that should be url-decoded when converting to a windows path?

Instead of using the original string + regex + string replace, `Path.GetFullPath(uri.AbsolutePath)`

https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUKgBgAJVMA6ASQHkBuPDIgBSIG88iOTMA2EgFiIBZAIYBLKAAoAlK3ad5AN2FgiEMKKIBeIlACmAdyIBVdRIBEAM1EAbXSAD0jgMIPgugM7B7AOQMBSdAARAE1dZTIAWzgzKTpceXlSYmsAewBjYWtGYWAACy0mHNyyAHFdYAAxCGssook1UTIAQQAjdxTrCDdsvNi5BK4ATglUjNreuPkAXzwpoA===

Reply:
I apologize, I missed the description and jumped to the code.

I would like to understand more why we have URL's that have encoded strings, but shouldn't be decoded. 

The original URLs are wrong. Where did they come from like that?